stdlib/src/builtin/string_literal.mojo

```diff
@@ -144,6 +145,14 @@ struct StringLiteral(
         """
         return self
 
+    fn __repr__(self) -> String:
```

Comment:
Please add tests for this, directly using this method (as opposed to the builting `repr` function).

Reply:
And please place them in `test_string_litera.mojo`.

Reply:
ok that sounds good. I'm also going to move the `repr` escape string tests into `test_string.mojo` (also applying the suggestion below), and add corresponding ones into `test_string_literal.mojo`.

Reply:
it looks like things are moving towards not calling them directly in the tests, so I'm still going to use the `repr` function

Reply:
Where do you see things not moving in that direction?